llpc/lower/llpcSpirvLowerGlobal.cpp

```diff
@@ -1790,6 +1792,13 @@ static Instruction *findNearestCommonDominator(Instruction *instA, Use &useB, Do
     instB = blockB->getTerminator();
   }
 
+  // Check if useB is actually reachable
+  if (!domTree.getNode(blockB)) {
+    unreachableUses.push_back(instB);
+    return instA;
+  }
+
+  usesToReplace.push_back(instB);
```

Comment:
Instead of doing this, I think we can switch to `Instruction *` version of `findNearestCommonDominator()`.
That is replace:
```
  BasicBlock *blockA = instA->getParent();
  if (blockA == blockB)
    return instA->comesBefore(instB) ? instA : instB;
  BasicBlock *block = domTree.findNearestCommonDominator(blockA, blockB);
  return block == blockA ? instA : (block == blockB ? instB : block->getTerminator());
```
with:
```
return domTree.findNearestCommonDominator(instA, instB);
```

Reply:
Thanks - that's a good point. See the latest proposal.